k8s-client.gemspec

```diff
@@ -21,18 +21,19 @@ Gem::Specification.new do |spec|
   spec.bindir        = "bin"
   spec.executables   = []
   spec.require_paths = ["lib"]
-  spec.required_ruby_version = '~> 2.4'
+#  spec.required_ruby_version = '> 2.4'
 
-  spec.add_runtime_dependency "excon", "~> 0.66"
-  spec.add_runtime_dependency "recursive-open-struct", "~> 1.1.0"
-  spec.add_runtime_dependency 'hashdiff', '~> 1.0.0'
-  spec.add_runtime_dependency 'jsonpath', '~> 0.9.5'
-  spec.add_runtime_dependency "yaml-safe_load_stream", "~> 0.1"
+  spec.add_runtime_dependency "excon"#, "~> 0.66"
+  spec.add_runtime_dependency "recursive-open-struct"#, "~> 1.1.0"
+  spec.add_runtime_dependency 'hashdiff'#, '~> 1.0.0'
+  spec.add_runtime_dependency 'jsonpath'#, '~> 0.9.5'
+  spec.add_runtime_dependency "yaml-safe_load_stream-renewed"#, "~> 0.1"
 
-  spec.add_development_dependency "bundler", ">= 1.17", "< 3.0"
-  spec.add_development_dependency "rake", "~> 10.0"
-  spec.add_development_dependency "rspec", "~> 3.7"
-  spec.add_development_dependency "webmock", "~> 3.6.2"
-  spec.add_development_dependency "rubocop", "~> 0.59"
-  spec.add_development_dependency 'yajl-ruby', '~> 1.4.0'
+#  spec.add_development_dependency "bundler"#, ">= 1.17", "< 3.0"
+  spec.add_development_dependency "byebug"#, "~> 10.0"
+  spec.add_development_dependency "rake"#, "~> 10.0"
+  spec.add_development_dependency "rspec"#, "~> 3.7"
+  spec.add_development_dependency "webmock"#, "~> 3.6.2"
+  spec.add_development_dependency "rubocop"#, "~> 0.59"
+  spec.add_development_dependency 'yajl-ruby'#, '~> 1.4.0'
```

Comment:
Which of these gemspec changes are necessary? Do you really want all dependencies listed with no version costrains?

Reply:
counter question: do you really want to drag along 3 year outdated dependencies although the checks are all green?

Reply:
I see, it's a useful experiment and good question for a gem that's not very active.  (My question was more "do these belong in same PR" or whether they were debug leftovers committed by mistake.)

However, if you drop constraints entirely, you also allow _even older_ versions!  Anything at all if it lets bundler solve the app constraints.
For dev dependencies, maybe that's fine.
For runtime dependencies, IMHO it's safer to at least set `>=`, and preferably pin major versions to current one.

- I never remember what exactly `~>` means, love [this reference](https://snyk.io/blog/differences-in-version-handling-gems-and-npm/#advanced-range-syntax) :eyes:...  OK, `~> 3.7` means 3.y (y >=7) and is useful; `~> 3.6.2` form also constrains minor 3.6.z (z >= 2) and is probably unnecessarily strict.

Anyway, I'm not a maintainer here, just watching and pushing my nose occasionally, these are more questions to @kke @jakolehm...

Reply:
I haven't touched ruby in a long time. I believe the same applies to many of the other past contributors. If any of you want to take ownership and maintain the gem, I think we will gladly transfer/grant that, @matti ?

Reply:
we tried to move the development in this fork: https://github.com/k8s-ruby/k8s-ruby

Reply:
@kke what about https://github.com/kontena/k8s-client/pull/175